mlflow/pyfunc/__init__.py

```diff
@@ -2268,7 +2281,14 @@ def predict(model_input: List[str]) -> List[str]:
     """
     _validate_env_arguments(conda_env, pip_requirements, extra_pip_requirements)
     _validate_pyfunc_model_config(model_config)
+    _validate_and_prepare_target_save_path(path)
+
     if python_model:
+        model_code_path = None
+        if isinstance(python_model, str):
+            model_code_path = _validate_and_get_model_code_path(python_model)
+            _validate_and_copy_model_code_and_config_paths(model_code_path, None, path)
+            python_model = _load_model_code_path(model_code_path)
```

Comment:
The exception message raised in line 2297 https://github.com/mlflow/mlflow/pull/11901/files#diff-592813f9fb198dda0982794f2221a408f316709190b3bf02c0e77eb2c1107075R2297 may be confusing if python_model is passed in as a string